### PR TITLE
Add spring rest docs, test

### PR DESCRIPTION
**Pull request by shalauniou**

### CODE REVIEW

File: src/test/java/com/epam/petclinic/clinic/controller/ClinicControllerTest.java [hunk 1]
```diff
@@ -57,7 +57,7 @@
 
     @Test
     public void testGetAllClinics() {
-        List<Clinic> clinics = restTemplate.getForObject(getHost() + CLINIC_PATH, List.class);
+        List clinics = restTemplate.getForObject(getHost() + CLINIC_PATH, List.class);
```
Comment (Andrei-Khadziukou) [inline src/test/java/com/epam/petclinic/clinic/controller/ClinicControllerTest.java]: What is a problem with parameterized list?

Comment (shalauniou) [reply src/test/java/com/epam/petclinic/clinic/controller/ClinicControllerTest.java]: It is highlighted in idea and  build log has
Note: D:\nasdaq\spring-cloud\petclinic-clinic\src\test\java\com\epam\petclinic\clinic\controller\ClinicControllerTest.java uses unchecked or unsafe operations.

File: src/main/java/com/epam/petclinic/clinic/model/Animal.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 package com.epam.petclinic.clinic.model;
 
 import groovy.transform.EqualsAndHashCode;
-import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang.builder.ToStringBuilder;
```
Comment (Andrei-Khadziukou) [inline src/main/java/com/epam/petclinic/clinic/model/Animal.java]: I think it should be lang3

File: src/test/java/com/epam/petclinic/clinic/controller/ClinicControllerTest.java (hunk 1 above)
Comment (Andrei-Khadziukou) [reply src/test/java/com/epam/petclinic/clinic/controller/ClinicControllerTest.java]: Could you check whether `List<?>` helps?

File: src/main/java/com/epam/petclinic/clinic/controller/ClinicController.java [hunk 3]
```diff
@@ -37,7 +37,9 @@
     @RequestMapping(value = "/", method = RequestMethod.POST)
     public Clinic create(@RequestBody Clinic clinic) {
         List<Offer> offers = clinic.getOffers();
-        offers.stream().forEach(offer -> offer.setClinic(clinic));
+        if (CollectionUtils.isNotEmpty(offers)) {
```
Comment (Andrei-Khadziukou) [inline src/main/java/com/epam/petclinic/clinic/controller/ClinicController.java]: Do we need that to prevent NPE?

Comment (shalauniou) [reply src/main/java/com/epam/petclinic/clinic/controller/ClinicController.java]: Yes, clinic could be created without offers

Comment (Andrei-Khadziukou) [reply src/main/java/com/epam/petclinic/clinic/controller/ClinicController.java]: Try to use Optional

File: build.gradle [hunk 4]
```diff
@@ -4,11 +4,13 @@ buildscript {
     }
     repositories {
         mavenCentral()
+        jcenter()
     }
     dependencies {
         classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
         classpath("org.liquibase:liquibase-gradle-plugin:${gradlePluginVersion}")
         classpath("org.postgresql:postgresql:${postgreSqlVersion}")
+        classpath("org.asciidoctor:asciidoctor-gradle-plugin:${asciidoctorPluginVersion}")
```
Comment (Andrei-Khadziukou) [inline build.gradle:13]: Could you please do a research whether there the way to manage such dependencies's versions in BOM?

File: src/main/resources/bootstrap.yml [hunk 5]
```diff
@@ -11,9 +11,7 @@ eureka:
     leaseExperationDurationinSecond: 20
   client:
     serviceUrl:
-      defaultZone: http://127.0.0.1:8080/eureka/
-    healthcheck:
-      enabled: true
+      defaultZone: http://epbyvitw0013:8080/eureka/
```
Comment (Andrei-Khadziukou) [inline src/main/resources/bootstrap.yml]: localhost

**Pull request merged**